bioio_ome_zarr/writers/ome_zarr_writer.py

```diff
@@ -0,0 +1,489 @@
+                    raise ValueError(
+                        f"Each scale tuple must have length {self.ndim}; "
+                        f"got {len(level_scale)}"
+                    )
+
+            # Normalize to the declared type (List[List[float]])
+            self.dataset_scales = [list(map(float, tpl)) for tpl in scale]
+            for vec in self.dataset_scales:
+                next_shape = tuple(
+                    max(1, int(np.floor(self.shape[i] * vec[i])))
+                    for i in range(self.ndim)
+                )
+                if next_shape == self.level_shapes[-1]:
+                    continue
+                self.level_shapes.append(next_shape)
+
+        self.num_levels = len(self.level_shapes)
+
+        # 4) Determine per-level chunk shapes
+        self._chunk_shape_explicit = chunk_shape is not None
+        if chunk_shape is not None:
+            # Detect single-tuple vs tuple-of-tuples
+            if len(chunk_shape) > 0 and isinstance(chunk_shape[0], int):
+                single = cast(ChunkSingle, chunk_shape)
+                if len(single) != self.ndim:
+                    raise ValueError(
+                        f"chunk_shape length {len(single)} != ndim {self.ndim}"
+                    )
+                self.chunk_shapes_per_level = [
+                    tuple(int(x) for x in single) for _ in range(self.num_levels)
+                ]
+            else:
+                per_level = cast(ChunkPerLevel, chunk_shape)
+                if len(per_level) != self.num_levels:
+                    raise ValueError(
+                        f"chunk_shape must match levels: ({self.num_levels}); "
+                        f"got {len(per_level)}"
+                    )
+                shapes: List[Tuple[int, ...]] = []
+                for idx, ch in enumerate(per_level):
+                    if len(ch) != self.ndim:
+                        raise ValueError(
+                            f"chunk_shape[{idx}] length {len(ch)} != ndim {self.ndim}"
+                        )
+                    shapes.append(tuple(int(x) for x in ch))
+                self.chunk_shapes_per_level = shapes
+        else:
+            suggested = chunk_size_from_memory_target(
+                self.level_shapes[0], self.dtype, 16 << 20
+            )
+            self.chunk_shapes_per_level = [suggested for _ in range(self.num_levels)]
+
+        # 5) formatting and compression
+        self.zarr_format = zarr_format
+        self.shard_factor = shard_factor
+        self.compressor = compressor
+
+        # 6) Metadata fields
+        self.image_name = image_name or "Image"
+        self.channels = channels
+        self.rdefs = rdefs
+        self.creator_info = creator_info
+        self.root_transform = root_transform
+
+        # 8) Handles & state
+        self.root: Optional[zarr.Group] = None
+        self.datasets: List[zarr.Array] = []
+        self._initialized: bool = False
```

Comment:
These could just be defaulted to these states rather than needing to be set

Reply:
resolved in [2fbb5d9](https://github.com/bioio-devs/bioio-ome-zarr/pull/72/commits/2fbb5d988111130afe71b1a414e9c37699dfdfdb)